tests/e2e/helpers/loadExtension.ts

```diff
@@ -149,23 +155,83 @@ export async function loginToExistingAlbyAccount(page: Page) {
     password: "12345678",
   };
 
-  const loginButton = await getByText($document, "Log in");
-  loginButton.click();
+  const connectButton = await getByText($document, "Connect with Alby");
+  connectButton.click();
+
+  const currentTarget = await page.target();
+  console.info("Current target: " + currentTarget.url());
+
+  //check that the first page opened this new page:
+  const newTarget = await page
+    .browser()
+    .waitForTarget(
+      (target) => target.url().indexOf("app.regtest.getalby.com") > -1,
+      {
+        timeout: 20000,
+      }
+    );
+  console.info("Found target: " + newTarget.url());
+  //get the new page object:
+  const oauthPage = await newTarget.page();
+  if (!oauthPage) {
+    throw new Error("OAuth page not found");
+  }
+  const oauthDocument = await getDocument(oauthPage);
 
-  await findByText($document, "Your Alby Account");
+  const oauthLoginButton = await getByText(oauthDocument, "Log in to connect");
+  oauthLoginButton.click();
 
-  // type user email
-  const emailField = await getByLabelText(
-    $document,
-    "Email Address or Lightning Address"
+  await oauthPage.waitForNavigation();
+  const oauthDocument2 = await getDocument(oauthPage);
+
+  const oauthLoginPasswordButton = await getByText(
+    oauthDocument2,
+    "Log in with password"
   );
+  oauthLoginPasswordButton.click();
+
+  await oauthPage.waitForNavigation();
+  const oauthDocument3 = await getDocument(oauthPage);
+
+  await new Promise((resolve) => setTimeout(resolve, 1000));
+
+  // type user email
+  const emailField = await getByLabelText(oauthDocument3, "Email address");
   await emailField.type(user.email);
 
   // type user password and confirm password
-  const walletPasswordField = await getByLabelText($document, "Password");
+  const walletPasswordField = await getByLabelText(oauthDocument3, "Password");
   await walletPasswordField.type(user.password);
 
-  await commonCreateWalletSuccessCheck({ page, $document });
+  const oauthConfirmLoginButton = await getByText(oauthDocument3, "Log in");
+  oauthConfirmLoginButton.click();
+
+  await oauthPage.waitForNavigation();
+  const oauthDocument4 = await getDocument(oauthPage);
+
+  const oauthConfirmAuthButton = await getByText(
+    oauthDocument4,
+    "Connect with Alby Extension"
+  );
+  oauthConfirmAuthButton.click();
+
+  let retries = 0;
+  const MAX_RETRIES = 20;
```

Comment:
Why would we want to retry here? Is there some reason it wouldn't work at first, but then it would after retrying?

Reply:
The variable maybe could be renamed, hmm. What it's doing is waiting for the OAuth modal to close, and the redirection to occur. I couldn't find a better way to do it.